pandas/io/parsers.py

```diff
@@ -40,70 +48,79 @@ class ParserWarning(Warning):
+header : int or list of ints, default 'infer'
     Row number(s) to use as the column names, and the start of the data.
-    Defaults to 0 if no ``names`` passed, otherwise ``None``. Explicitly pass
-    ``header=0`` to be able to replace existing names. The header can be a list
-    of integers that specify row locations for a multi-index on the columns
-    E.g. [0,1,3]. Intervening rows that are not specified will be skipped
-    (e.g. 2 in this example are skipped). Note that this parameter ignores
-    commented lines and empty lines if ``skip_blank_lines=True``, so header=0
-    denotes the first line of data rather than the first line of the file.
-skiprows : list-like or integer, default None
-    Line numbers to skip (0-indexed) or number of lines to skip (int)
-    at the start of the file
+    Default behavior is as if set to 0 if no ``names`` passed, otherwise
+    ``None``. Explicitly pass ``header=0`` to be able to replace existing names.
+    The header can be a list of integers that specify row locations for a
+    multi-index on the columns e.g. [0,1,3]. Intervening rows that are not
+    specified will be skipped (e.g. 2 in this example is skipped). Note that
+    this parameter ignores commented lines and empty lines if
+    ``skip_blank_lines=True``, so header=0 denotes the first line of data rather
+    than the first line of the file.
+names : array-like, default None
+    List of column names to use. If file contains no header row, then you
+    should explicitly pass header=None
 index_col : int or sequence or False, default None
     Column to use as the row labels of the DataFrame. If a sequence is given, a
     MultiIndex is used. If you have a malformed file with delimiters at the end
     of each line, you might consider index_col=False to force pandas to _not_
     use the first column as the index (row names)
-names : array-like, default None
-    List of column names to use. If file contains no header row, then you
-    should explicitly pass header=None
-prefix : string, default None
-    Prefix to add to column numbers when no header, e.g 'X' for X0, X1, ...
-na_values : str, list-like or dict, default None
-    Additional strings to recognize as NA/NaN. If dict passed, specific
-    per-column NA values
+usecols : array-like, default None
+    Return a subset of the columns.
+    Results in much faster parsing time and lower memory usage.
+squeeze : boolean, default False
+    If the parsed data only contains one column then return a Series
+prefix : str, default None
+    Prefix to add to column numbers when no header, e.g. 'X' for X0, X1, ...
+mangle_dupe_cols : boolean, default True
+    Duplicate columns will be specified as 'X.0'...'X.N', rather than 'X'...'X'
+dtype : Type name or dict of column -> type, default None
+    Data type for data or columns. E.g. {'a': np.float64, 'b': np.int32}
+    (Unsupported with engine='python'). Use `str` or `object` to preserve and
+    not interpret dtype.
+%s
```

Comment:
engine: gets filled in here. It's tricky to use ''.format() because of the multiple dictionary examples in the docstring containing curly braces. I wanted to call attention to this because it does make the docstring a bit more complicated. The reason it's dynamic is so it is left out of the read_fwf() docs (existing behavior)